app/src/main/java/com/nervousfish/nervousfish/activities/NFCActivity.java

```diff
@@ -1,22 +1,147 @@
 package com.nervousfish.nervousfish.activities;
 
 import android.app.Activity;
+import android.content.Intent;
+import android.nfc.NdefMessage;
+import android.nfc.NdefRecord;
+import android.nfc.NfcAdapter;
+import android.nfc.NfcAdapter.CreateNdefMessageCallback;
+import android.nfc.NfcEvent;
 import android.os.Bundle;
+import android.os.Parcelable;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
 
+import com.bumptech.glide.Glide;
+import com.nervousfish.nervousfish.ConstantKeywords;
 import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Contact;
+import com.nervousfish.nervousfish.data_objects.Profile;
+import com.nervousfish.nervousfish.data_objects.SimpleKey;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static android.nfc.NdefRecord.createExternal;
 
 /**
- * An {@link Activity} that used for pairing over NFC
+ * An {@link Activity} that beams NDEF Messages to Other Devices.
  */
-public class NFCActivity extends Activity {
+public class NFCActivity extends Activity implements CreateNdefMessageCallback {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("NFCActivity");
+    private TextView textView;
+    private IServiceLocator serviceLocator;
+    private byte[] bytes;
+
     /**
-     * Creates the new activity, should only be called by Android
-     *
-     * @param savedInstanceState The saved state of the instance.
+     * {@inheritDoc}
      */
     @Override
-    protected void onCreate(final Bundle savedInstanceState) {
+    public void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_nfc);
+
+        final Intent intent = getIntent();
```

Comment:
Please download https://www.dropbox.com/s/wryiggq2x9yr0tz/Project_Default.xml?dl=0
Press Ctrl-alt-s -> Editor -> Inspections -> Manage -> Import
Hover over getIntent and press Cleanup code